app/react/Library/components/PDFUploadButton.tsx

```diff
@@ -0,0 +1,99 @@
+import { Translate } from 'app/I18N';
+import { wrapDispatch } from 'app/Multireducer';
+import { Icon } from 'app/UI';
+import React, { ChangeEvent, Dispatch } from 'react';
+import { bindActionCreators } from 'redux';
+import { EntitySchema } from 'shared/types/entityType';
+
+import {
+  uploadDocument as uploadDocumentAction,
+  createDocument as createDocumentAction,
+} from 'app/Uploads/actions/uploadsActions';
+
+import { unselectAllDocuments as unselectAllDocumentsAction } from 'app/Library/actions/libraryActions';
+import { connect } from 'react-redux';
+
+const extractTitle = (file: File) => {
+  const title = file.name
+    .replace(/\.[^/.]+$/, '')
+    .replace(/_/g, ' ')
+    .replace(/-/g, ' ')
+    .replace(/ {2}/g, ' ');
+
+  return title.charAt(0).toUpperCase() + title.slice(1);
+};
+
+interface PDFUploadActions {
+  createDocument: (e: EntitySchema) => any;
+  uploadDocument: (s: string, f: File) => void;
+  unselectAllDocuments: () => void;
+}
+
+interface PDFUploadButtonOwnProps {
+  storeKey: string;
+}
+
+type PDFUploadButtonProps = PDFUploadActions & PDFUploadButtonOwnProps;
+
+const onChangePDFs = ({
+  createDocument,
+  uploadDocument,
+  unselectAllDocuments,
+}: PDFUploadActions) => (e: ChangeEvent<HTMLInputElement>) => {
+  const { files } = e.target as HTMLInputElement;
+  const count = files?.length || 0;
+
+  // eslint-disable-next-line no-plusplus
+  for (let index = 0; index < count; index++) {
+    const file = files![index];
+    const doc = { title: extractTitle(file) };
+    createDocument(doc)
+      .then((newDoc: EntitySchema) => {
+        uploadDocument(newDoc.sharedId!, file);
+      })
+      .catch(() => {});
+  }
+
+  unselectAllDocuments();
+};
+
+const PDFUploadButtonComponent = ({
+  createDocument,
+  uploadDocument,
+  unselectAllDocuments,
+}: PDFUploadButtonProps) => (
+  <label htmlFor="pdf-upload-button" className="btn btn-success">
+    <Icon icon="cloud-upload-alt" />
+    <span className="btn-label">
+      <Translate>Upload PDF(s) to create</Translate>
+    </span>
+    <input
+      type="file"
+      id="pdf-upload-button"
+      style={{ display: 'none' }}
+      accept="application/pdf"
+      multiple
+      onChange={onChangePDFs({
```

Comment:
I think this will do unnecessary re-renders.

Reply:
This code is extracted from the previous implementation on the uploads screen. Is not revised or refactored. 
Would it make sense to check for re-renders?

Reply:
@Zasa-san @konzz my bad, actually we did write that piece of code. We will take a look.